ping.go

```diff
@@ -442,7 +486,7 @@ func (p *Pinger) recvICMP(
 func (p *Pinger) processPacket(recv *packet) error {
 	receivedAt := time.Now()
 	var proto int
-	if p.ipv4 {
+	if *p.ipv4 {
```

Comment:
Is there any chance any of these if-statements could be reached with p.ipv4 still being unset? Just for the sake of safety could you change these to:

```
if p.ipv4 == nil || *p.ipv4 {
    ...
}
```

Reply:
We have to pass through a `Resolve()` cleanly which will always set the packet ipv4.

If `p.ipv4` were nil at this point, I would probably throw an error or `panic()`.

Reply:
it will panic already when you try to dereference a nil pointer. That's what I want to protect against, I don't want it to panic because libraries generally never should.

Reply:
But it's completely invalid to continue here if `p.ipv4 == nil`, which is what your change is suggesting.

Reply:
I was thinking about another way to solve this concern. Rather than depend on the set/unset state of the ipv4, we can have a second `resolved` bool in the Pinger struct.

Reply:
you could also make it an enum, something like:
```go
type networkType int

const (
	unset networkType = iota
	ipv4
	ipv6
)
```
I realize it's not valid for it to be unset, but the current behavior is for there to be a defacto 'default' behavior and NOT to panic. This library should avoid panics at all costs.

Reply:
Hrm, that's an interesting idea. I'll think about the options and spend some time on it soon. I'm starting to deploy my smokeping tool to production and it would be good to be able to better control protocol selection. :smile: